jni/include/native_engines_stream_support.h

```diff
@@ -119,6 +121,110 @@ class NativeEngineIndexInputMediator {
 
 
 
+/**
+ * This class delegates the provided index output to do IO processing.
+ * In most cases, it is expected that IndexOutputWithBuffer was passed down to this,
+ * which eventually have Lucene's IndexOutput to write bytes.
+ */
+class NativeEngineIndexOutputMediator {
+ public:
+  NativeEngineIndexOutputMediator(JNIUtilInterface *_jni_interface,
```

Comment:
Could we improve readability and safety by validating input parameters. Check for null pointers for jni_interface, env, and indexOutput to prevent potential issues.

Reply:
em.. I can, but is it needed? :sweat_smile: 
I've added unit tests to cover this, and there will be a single path to pass pointer which is created in stack area.
It is guaranteed non-null for all parameters including JNIUtilInterface.
but, if you insist, then will update in the next rev!

Any thoughts?

Reply:
Will leave up to you !!